### DISCUSSION TITLE
If Hy is the Python AST in Lisp form, couldn't that same concept be applied to any other language with an AST?

### DISCUSSION THREAD

**Discussion by willhansen** [hylang/hy#2611] [General]
Are there existing projects in that direction?  

There's no way I'm the first one to think of this.

**Comment by Kodiologist** [comment]
I've seen several projects that implement a sort of Lisp syntax for another programming language, as with [Axel](https://axellang.github.io) for Haskell, and [lots of different ones](https://github.com/jashkenas/coffeescript/wiki/List-of-languages-that-compile-to-JS#lisp-scheme) for JavaScript. But I don't know how common it is to use the host language's AST objects as part of the translation. That's a pretty fiddly implementation detail.

**Reply by willhansen** [reply]
It's important that it bypasses the default syntax rather than just transpiling.

I'm thinking along the lines of a general way to separate language features from specific syntax.  To avoid bikeshedding, if nothing else.

**Comment by addisaden** [comment]
Well that's an intresting idea - Clojure has a similar idea, but don't use the ast. It compiles to the jvm (default), javascript (clojurescript) and there were some other ideas to implement it on other platforms. What they are doing today, I don't know

**Reply by atisharma** [reply]
It's most common to use a vm target like you point to, I think. LFE, Clojerl and Elixir compile to the Erlang VM (BEAM) like Clojure on the JVM.

The basilisp reader/compiler [constructs an AST](https://basilisp.readthedocs.io/en/latest/compiler.html) then produces intermediate python code using `ast`.
Hissp [produces its own ast that compiles to python code](https://hissp.readthedocs.io/en/v0.2.0/faq.html#i-got-here-from-a-link-in-hy-s-docs-what-are-the-main-differences-between-hissp-and-hy)
